extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -309,12 +308,16 @@ private ShuffleSpec findShuffleSpecForNextWindow(List<OperatorFactory> operatorF
       }
     }
 
-    if (partition == null || partition.getPartitionColumns().isEmpty()) {
+    if (partition == null) {
```

Comment:
Do we have any tests which cover `partition == null`? Would the same bug not be present here?

Reply:
We don't have any tests currently for most of the window function specific files. I plan on adding them post the major refactoring work is done. We do have drill tests where I try to add any new case I stumble upon.

But for now, `partition == null` condition is correct. We do have drill tests to verify that flow.

`partition == null` condition is different than `partition.getPartitionColumns().isEmpty()`.

`partition == null` indicates that the group of operators doesn't have any NaivePartitioningOperatorFactory, whereas the latter one indicates that we need to bring all data together for computing an empty over clause.